src/services/transactions_details.rs

```diff
@@ -99,36 +98,56 @@ pub fn get_transactions_details(
     context: &Context,
     details_id: &String,
 ) -> ApiResult<TransactionDetails> {
+    let id_parts = parse_id(details_id)?;
+
+    match id_parts {
+        TransactionIdParts::Ethereum {
+            safe_address,
+            transaction_hash,
+            details_hash
+        } => get_ethereum_transaction_details(context, &safe_address, &transaction_hash, &details_hash),
+        TransactionIdParts::Module {
+            safe_address,
+            transaction_hash,
+            details_hash
+        } => get_module_transaction_details(context, &safe_address, &transaction_hash, &details_hash),
+        TransactionIdParts::Multisig { safe_tx_hash, .. } => get_multisig_transaction_details(context, &safe_tx_hash),
+        TransactionIdParts::TransactionHash(safe_tx_hash) => get_multisig_transaction_details(context, &safe_tx_hash),
+        _ => Err(ApiError::new_from_message(String::from("Bad transaction id"))),
+    }
+}
+
+pub(super) fn parse_id(details_id: &str) -> Result<TransactionIdParts> {
```

Comment:
Maybe that should be moved to a more general module, as this might be used in multiple places (e.g. submit confirmations, right?

Reply:
In the end I just ask for the `safeTxHash` as it is a more "common" way between the backend and the gateway to identify transactions. I was thinking that if we ever want to implement confirmation through notifications (not saying it's a good idea, just fantasising here :) ) It would be better to keep things flexible.